File: backend/app/schemas/ai.py
"""AI 相关响应模式"""

from pydantic import BaseModel


class ChatResponse(BaseModel):
    """AI 对话响应"""
    reply: str


class GenerateStepsResponse(BaseModel):
    """AI 生成测试步骤响应"""
    steps: list[dict]


class AnalyzePageResponse(BaseModel):
    """AI 页面分析响应"""
    analysis: str


class ChatRequest(BaseModel):
    """AI 对话请求"""
    message: str
    conversation_id: int | None = None


class GenerateStepsRequest(BaseModel):
    """AI 生成步骤请求"""
    description: str


class AnalyzePageRequest(BaseModel):
    """AI 页面分析请求"""
    url: str

File: frontend/src/services/ai.ts
// AI 助手 API 服务
import request from './request'

export interface ChatReq {
  message: string
  conversation_id?: number
}

export interface ChatRes {
  reply: string
}

export interface GenerateStepsReq {
  description: string
}

export interface TestStep {
  seq: number
  action: string
  target: string
  value?: string
  expected: string
}

export interface GenerateStepsRes {
  steps: TestStep[]
}

export interface AnalyzePageReq {
  url: string
}

export interface AnalyzePageRes {
  analysis: string
}

// AI 对话
export function aiChat(data: ChatReq): Promise<ChatRes> {
  return request.post('/ai/chat', data)
}

// AI 生成测试步骤
export function generateSteps(data: GenerateStepsReq): Promise<GenerateStepsRes> {
  return request.post('/ai/generate-steps', data)
}

// AI 分析页面
export function analyzePage(data: AnalyzePageReq): Promise<AnalyzePageRes> {
  return request.post('/ai/analyze-page', data)
}
